tests/testthat/test-variable-as-methods.R

```diff
@@ -0,0 +1,228 @@
+        expect_is(var_def, "CrunchExpr")
+        expect_equal(var_def, zfuncExpr("parse_datetime", ds$textVar, 
+                                        list(value = "%Y")))
+        
+        expect_error(ds$textVar <- var_def, save_over_error, fixed = TRUE)
+    })
+})
+
+with_test_authentication({
+    ds <- newDataset(df)
+
+    test_that("numeric to text", {
+        ds$v1_text <- as.Text(ds$v1)
+        expect_true(is.derived(ds$v1_text))
+        expect_true(is.Text(ds$v1_text))
+        expect_equal(as.vector(ds$v1_text), 
+                     as.character(as.vector(ds$v1)))
+        
+        # alias to R's function names
+        expect_equal(as.character(ds$v1), as.Text(ds$v1))
+    })
+    
+    test_that("text to numeric", {
+        # make a text variable with numbers
+        ds$num_as_text <- as.character(c(1:20))
+        
+        ds$num_as_text_num <- as.Numeric(ds$num_as_text)
+        expect_true(is.derived(ds$num_as_text_num))
+        expect_true(is.Numeric(ds$num_as_text_num))
+        expect_equal(as.vector(ds$num_as_text_num), 
+                     as.numeric(as.vector(ds$num_as_text)))
+        
+        # if there aren't numbers in the text, the result should be NA
+        ds$v2_num <- as.Numeric(ds$v2)
+        expect_true(is.derived(ds$v2_num))
+        expect_true(is.Numeric(ds$v2_num))
+        expect_equal(as.vector(ds$v2_num), 
+                     expect_warning(as.numeric(as.vector(ds$v2)),
+                     "NAs introduced by coercion"))
+        
+        # alias to R's function names
+        expect_equal(as.numeric(ds$v1), as.Numeric(ds$v1))
+    })   
+
+    test_that("text to categorical", {
+        ds$v2_cat <- as.Categorical(ds$v2)
+        expect_true(is.derived(ds$v2_cat))
+        expect_true(is.Categorical(ds$v2_cat))
+        expect_equal(as.vector(ds$v2_cat), 
+                     as.factor(as.vector(ds$v2)))
+    })  
+    
+    test_that("datetime to numeric", {
+        ds$v5_num <- as.Numeric(ds$v5)
+        expect_true(is.derived(ds$v5_num))
+        expect_true(is.Numeric(ds$v5_num))
+        expect_equal(as.vector(ds$v5_num), 
+                     as.numeric(as.vector(ds$v5)))
+    }) 
+    
+    test_that("datetime to text", {
+        ds$v5_text <- as.Text(ds$v5)
+        expect_true(is.derived(ds$v5_text))
+        expect_true(is.Text(ds$v5_text))
+        expect_equal(as.vector(ds$v5_text), 
+                     format(as.vector(ds$v5), format = "%Y-%m-%d %H:%M:%S"))
+    }) 
+    
+    test_that("datetime to text", {
+        ds$v5_cat <- as.Categorical(ds$v5)
+        expect_true(is.derived(ds$v5_cat))
+        expect_true(is.Categorical(ds$v5_cat))
+        expect_equal(as.vector(ds$v5_cat), 
+                     as.factor(
+                         format(as.vector(ds$v5), format = "%Y-%m-%d %H:%M:%S")
+                     ))
+    }) 
+    
+    
+    test_that("text to datetime", {
+        ds$text_times <- format(as.vector(ds$v5), format = "%Y-%m-%d")
```

Comment:
instead of `as.vector(ds$v5)`, you can just `df$v5` (the R version you uploaded)